pkg/storage/s3.go

```diff
@@ -10,39 +10,40 @@ const (
 	s3EndpointOption = "s3.endpoint"
 )
 
-// S3EndpointOption is a BackendOption for changing the endpoint of the storage.
-type S3Endpoint struct {
-	backendOption
-	// Endpoint is the host name of the S3 endpoint.
-	Endpoint string
+type S3BackendOptions struct {
+	Endpoint string `json:"endpoint" toml:"endpoint"`
 }
 
-func (o *S3Endpoint) applyOnS3(s3 *backup.S3, name string) error {
-	s3.Endpoint = o.Endpoint
+func (options *S3BackendOptions) validate() error {
 	return nil
 }
 
-// OptionName returns the name of this option.
-func (o *S3Endpoint) OptionName() string {
-	return s3EndpointOption
+func (options *S3BackendOptions) isEmpty() bool {
+	return options.Endpoint == ""
```

Comment:
Can we combine these two functions?
If we have 2 options for S3, both of them have default values, how can we determine if it's empty?